python/generate_syntax_highlightining.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+import json
+import os
+
+from pyoracc.atf.common.atflex import AtfLexer
+
+# In many cases we want to match both the lower-case "token" and the title-case
+# "Token".  This helper function creates a match for both options.
+def match_all_variants(token_names):
+    return "|".join(
+        [token.lower() for token in token_names]
+        + [token.title() for token in token_names]
+    )
+
+# Initialise the dictionary
+data = {}
+
+data["$schema"] = "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json"
+data["name"] = "ATF"
+data["patterns"] = [
+    {"include": "#keywords"},
+    {"include": "#support"},
+    {"include": "#strings"},
+    {"include": "#comment"},
+    {"include": "#variable"},
+]
+
+keywords = {}
+structures = match_all_variants(AtfLexer.structures)
+keywords["patterns"] = [
+    {"name": "keyword.control.at.atf",
+     "match": f"^@({structures})\\b"},
+    {"name": "keyword.other.ampersand.atf",
+     # For the sake of simplicity (and consistency with
+     # Nammu), highlight the entire &-line, whatever it is
+     # in there.
+     "match": "^&.*$"},
+]
+
+support = {}
+protocols = match_all_variants(AtfLexer.protocols)
+protocol_keywords = match_all_variants(AtfLexer.protocol_keywords)
+dollar_keywords = match_all_variants(AtfLexer.dollar_keywords)
+support["patterns"] = [
+    {"name": "support.class.hash.atf",
+     # Interlinear translations (`tr`) need to be treaded
+     # specially, because they're the only protocol allowed
+     # to be followed by something else (the specification
+     # of the language `.<LANG>`) before the colon.
+     "match": f"^#([Tt]r\\.[[:alpha:]]+|{protocols})\\b:[ \t]*({protocol_keywords})?"},
+    {"name": "support.variable.dollar.atf",
+     # We highlight the entire line starting with `$`, but we capture the known
+     # keywords.
+     "begin": "^\\$",
+     "end": "\\n",
+     "patterns": [
+         {
+             # At the moment we don't do anything special about these keywords,
+             # they're highlighted like the rest of the line, but we can
+             # capture them and give them a different name.
+             "name": "support.variable.dollar-keyword.atf",
```

Comment:
If we did want to highlight these in a different colour, would we need to give them a semantically-different name?

Reply:
That's my understanding as well